app/concerns/gkr_keyword_recommendations.rb

```diff
@@ -7,14 +7,24 @@ def fetch_keyword_recommendations(user, provider)
     response = cmr_client.fetch_keyword_recommendations(abstract)
 
     if response.error?
+      if response.status >= 500 && response.status <= 505    # NGAP LB can't connect to GKR.
+        log_gkr_comm_error(user, provider, abstract, response.status, 'Communication failure with GKR')
+        return { 'error': 'CommError' }
+      end
+
       log_gkr_request_failed(user, provider, abstract, response)
-      { id: '', recommendations: [] }
+      return { id: '', recommendations: [] }
     end
 
     keyword_recommendations = response.body['recommendations'].map { |rec| rec['keyword'] }
     { id: response.body['uuid'], recommendations: keyword_recommendations }
   end
 
+  def log_gkr_comm_error(user, provider, abstract, status, reason)
+    Rails.logger.info("GkrLog: type: FAILED - date: #{Time.new} - env: #{Rails.env} - user_id: #{user} - provider: #{provider}"\
```

Comment:
Can we include there request id that we are displaying to the user also in the log statements?   This way we can match them up?

Reply:
Ouch.  I should have caught that.  Will do.